apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPoolOptions.cs

```diff
@@ -0,0 +1,233 @@
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This property must always be positive. The default value is 15 minutes.
+        /// </para>
+        /// <para>
+        /// The exact value used is subject to "jitter" to avoid a lot of sessions being refreshed at the exact same time.
+        /// </para>
+        /// <para>
+        /// A lower value will cause sessions to be refreshed more often, slightly reducing the risk of sessions expiring
+        /// while being used, at the cost of performing more refreshes.
+        /// </para>
+        /// <para>
+        /// This value must be less than the expire timer on the Spanner server which is currently set at 60 minutes.
+        /// </para>
+        /// </remarks>
+        public TimeSpan IdleSessionRefreshDelay
+        {
+            get => _idleSessionRefreshDelay;
+            set => _idleSessionRefreshDelay = CheckPositiveTimeSpan(value);
+        }
+
+        /// <summary>
+        /// The amount of time before sessions are forcibly evicted from the pool. This is usually in the order of days,
+        /// as sessions can be reused for a long time if they're suitably refreshed. Deleting long-lasting sessions
+        /// can help with server-side resource management.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This property must always be positive. The default value is 7 days.
+        /// </para>
+        /// <para>
+        /// The exact value used is subject to "jitter" to avoid a lot of sessions being evicted at the exact same time.
+        /// </para>
+        /// <para>
+        /// A lower value will cause sessions to be evicted more often, delaying operations if no sessions are available
+        /// when requested.
+        /// </para>
+        /// </remarks>
+        public TimeSpan PoolEvictionDelay
+        {
+            get => _poolEvictionDelay;
+            set => _poolEvictionDelay = CheckPositiveTimeSpan(value);
+        }
+
+        /// <summary>
+        /// Determines the behavior of a request for a session when <see cref="MaximumActiveSessions"/> has been reached.
+        /// </summary>
+        /// <remarks>
+        /// The default value is <see cref="ResourcesExhaustedBehavior.Block"/>.
+        /// </remarks>
+        public ResourcesExhaustedBehavior WaitOnResourcesExhausted
+        {
+            get => _waitOnResourcesExhausted;
+            set => _waitOnResourcesExhausted = GaxPreconditions.CheckEnumValue(value, nameof(value));
+        }
+
+        /// <summary>
+        /// Fraction of sessions to be kept prepared for write transactions.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This is an optimisation to avoid the cost of sending a BeginTransaction() rpc.If all such sessions are in use and a
+        /// write request comes, we will make the BeginTransaction() rpc inline. The must be between 0 and 1 (inclusive).
+        /// </para>
```

Comment:
nit: The **value** must be betw...

Reply:
Removed the sentence entirely as it's covered in the next paragraph anyway.